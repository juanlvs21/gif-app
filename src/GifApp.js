import React, { useState } from "react";
import PropTypes from "prop-types";

import GifSearch from "./components/GifSearch";
import GifGrid from "./components/GifGrid";

const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifApp</h2>
      <GifSearch setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

GifApp.prototype = {
  defaultCategories: PropTypes.array,
};

export default GifApp;
