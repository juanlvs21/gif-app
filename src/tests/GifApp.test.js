import { shallow } from "enzyme";

import GifApp from "../GifApp";

describe("Pruebas en el componente GifSearch", () => {
  test("Cargar componente GifSearch correctamente", () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar una lista de categorias", () => {
    const categories = ["Tomioka", "Senku"];

    const wrapper = shallow(<GifApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
