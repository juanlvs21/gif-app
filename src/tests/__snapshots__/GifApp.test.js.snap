// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifSearch Cargar componente GifSearch correctamente 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <GifSearch
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente GifSearch Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <GifSearch
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Tomioka"
      key="Tomioka"
    />
    <GifGrid
      category="Senku"
      key="Senku"
    />
  </ol>
</Fragment>
`;
